base/src/main/java/proguard/evaluation/stateTrackers/HumanPrinter.java

```diff
@@ -0,0 +1,316 @@
+package proguard.evaluation.stateTrackers;
+
+import proguard.classfile.Clazz;
+import proguard.classfile.Method;
+import proguard.classfile.attribute.CodeAttribute;
+import proguard.classfile.attribute.ExceptionInfo;
+import proguard.classfile.instruction.Instruction;
+import proguard.classfile.instruction.InstructionFactory;
+import proguard.classfile.util.ClassUtil;
+import proguard.classfile.visitor.ClassPrinter;
+import proguard.evaluation.BasicBranchUnit;
+import proguard.evaluation.PartialEvaluator;
+import proguard.evaluation.TracedStack;
+import proguard.evaluation.TracedVariables;
+import proguard.evaluation.Variables;
+import proguard.evaluation.value.InstructionOffsetValue;
+
+import java.util.List;
+
+public class HumanPrinter implements PartialEvaluatorStateTracker
+{
+    private final boolean printDebugInfo;
+    private final boolean printDebugResults;
+
+    public HumanPrinter(boolean printDebugInfo, boolean printDebugResults)
+    {
+        this.printDebugInfo=printDebugInfo;
+        this.printDebugResults=printDebugResults;
+    }
+
+    // Code attribute level:
```

Comment:
what this mean?

Reply:
I would like to separate the file a little, by substates... This is the aim. How would you suggest I do this? Or maybe @mrjameshamilton has an idea?